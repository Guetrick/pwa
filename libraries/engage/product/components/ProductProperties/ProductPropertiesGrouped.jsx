import React from 'react';
import PropTypes from 'prop-types';
import { Accordion } from '@shopgate/pwa-ui-material';
import ProductPropertiesLists from './ProductPropertiesLists';
import ProductPropertiesWrapper from './ProductPropertiesWrapper';
import { accordion } from './ProductProperties.style';

/**
 * Renders the properties as groups.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const ProductPropertiesGrouped = ({ properties, groups }) => groups
  .map(group => (
    <div key={group} className={accordion}>
      <Accordion renderLabel={() => group} testId={`product-properties-group-${group}`}>
        <ProductPropertiesWrapper dense>
          <ProductPropertiesLists
            properties={properties.filter(property => property.displayGroup === group)}
          />
        </ProductPropertiesWrapper>
      </Accordion>
    </div>
  ));

ProductPropertiesGrouped.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.string).isRequired,
  properties: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default ProductPropertiesGrouped;
