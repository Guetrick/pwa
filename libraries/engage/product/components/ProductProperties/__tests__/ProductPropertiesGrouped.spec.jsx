import React from 'react';
import { shallow } from 'enzyme';
import ProductPropertiesGrouped from '../ProductPropertiesGrouped';

const properties = [
  {
    name: 'test1',
    label: 'Test 1',
    displayGroup: 'Group 1',
  },
  {
    name: 'test2',
    label: 'Test 2',
    displayGroup: 'Group 1',
  },
  {
    name: 'test3',
    label: 'Test 3',
    displayGroup: 'Group 2',
  },
  {
    name: 'test4',
    label: 'Test 4',
    displayGroup: 'Group 2',
  },
];

const groups = ['Group 1', 'Group 2'];

describe('<ProductPropertiesGrouped />', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ProductPropertiesGrouped properties={properties} groups={groups} />);
    expect(wrapper.find('Accordion').length).toEqual(2);
    expect(wrapper.find('ProductPropertiesWrapper').length).toEqual(2);
    expect(wrapper.find('ProductPropertiesWrapper').at(0).prop('dense')).toEqual(true);
    expect(wrapper.find('ProductPropertiesLists').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});
