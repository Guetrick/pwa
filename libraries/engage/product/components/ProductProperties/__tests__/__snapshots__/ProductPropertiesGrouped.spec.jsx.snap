// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesGrouped /> should render as expected 1`] = `
Array [
  <div
    className={
      Object {
        "data-css-fz7184": "",
      }
    }
    key="Group 1"
  >
    <Accordion
      renderLabel={[Function]}
      testId="product-properties-group-Group 1"
    >
      <ProductPropertiesWrapper
        dense={true}
      >
        <ProductPropertiesLists
          properties={
            Array [
              Object {
                "displayGroup": "Group 1",
                "label": "Test 1",
                "name": "test1",
              },
              Object {
                "displayGroup": "Group 1",
                "label": "Test 2",
                "name": "test2",
              },
            ]
          }
        />
      </ProductPropertiesWrapper>
    </Accordion>
  </div>,
  <div
    className={
      Object {
        "data-css-fz7184": "",
      }
    }
    key="Group 2"
  >
    <Accordion
      renderLabel={[Function]}
      testId="product-properties-group-Group 2"
    >
      <ProductPropertiesWrapper
        dense={true}
      >
        <ProductPropertiesLists
          properties={
            Array [
              Object {
                "displayGroup": "Group 2",
                "label": "Test 3",
                "name": "test3",
              },
              Object {
                "displayGroup": "Group 2",
                "label": "Test 4",
                "name": "test4",
              },
            ]
          }
        />
      </ProductPropertiesWrapper>
    </Accordion>
  </div>,
]
`;
